File: packages/vfuse-core/src/examples/javascript/PiEstimation.js
function getPoints(interval){
    let rand_x, rand_y, origin_dist, pi
    let circle_points = 0, square_points = 0

    for(let i =0; i < (interval * interval); i++){
        rand_x = (Math.random()* interval) / interval
        rand_y = (Math.random()* interval) / interval
        origin_dist = rand_x * rand_x + rand_y * rand_y
        square_points++
        if(origin_dist < 1)
            circle_points++
    }

    return {square_points : square_points, circle_points : circle_points}
}

function estimatePi(data){
    let circle_points = 0, square_points = 0
    for(let points of data){
        square_points += points.square_points
        circle_points += points.circle_points
    }

    pi = parseFloat((4 * circle_points) / square_points).toFixed(100)

    return pi
}

let interval = 10000
let step = 1000
for(let i = step; i <= interval; i+=step)
    await VFuse.addJob(getPoints, [], step)


await VFuse.addJob(estimatePi, ['getPoints'])

File: packages/vfuse-core/src/examples/python/PIEstimation.py
def monteCarloPiPart(n):
    import random
    count = 0
    for i in range(n):
        x=random.random()
        y=random.random()
        # if it is within the unit circle
        if x*x + y*y <= 1:
            count=count+1
    return count

def estimatePi(data):
    n = len(data) * 10000
    return sum(data)/(n*1.0)*4

jid = await VFuse.addJob(monteCarloPiPart, [], 10000)
await VFuse.setRepeating(jid)
jid = await VFuse.addJob(estimatePi, [jid])
await VFuse.setRepeating(jid)
